admin/win/tools/NCNavRemove/dll/dllmain.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) by Michael Schuster <michael.schuster@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "NCTools.h"
+#include "SimpleMutex.h"
+#include "NavRemoveConstants.h"
+
+SimpleMutex g_mutex;
```

Comment:
Had the same feeling but didn't know how to do it otherwise. I mean down to the bone it's the same as an App singleton. Shouldn't be an issue for this small DLL project ;)